### PR TITLE
fix: send a single RDB_OPCODE_FULLSYNC_END from a snapshot

### LINKED ISSUE #917: Replication sending single byte gets stuck and does not arrive to replica
When creating a replica to master, sending of single byte gets stuck and does not arrives to replica. To overcome this issue we currently send full sync cut 10 times.

**To Reproduce**
1. edit the code in file snapshot.cc to call serializer_->SendFullSyncCut() only once (now it is inside for loop)
2. start 2 instances of dragonfly and connect by cli
./dragonfly --alsologtostderr --vmodule=snapshot=2
./dragonfly --alsologtostderr --port 6380 --vmodule=replica=2
3. run cli : redis-cli -p 6380
4. run in cli:  REPLICAOF localhost 6379  

In the master logs you will see:
 snapshot.cc:236] PushSerializedToChannel 1 bytes
These are the prints of sending full sync cut to channel

In replica logs you will see:
replica.cc:541] Waiting for all full sync cut confirmations

Also if you run in replica cli:
get x
you will get the error : (error) ERR Can not execute during LOADING
So replica did not finished loading and did not get the full sync cut data

### PR DESCRIPTION
**Pull request by romange**
Fixes #917

### CODE REVIEW

File: src/server/rdb_save.cc [hunk 1]
```diff
@@ -666,7 +666,14 @@ error_code RdbSerializer::SaveStreamConsumers(streamCG* cg) {
 }
 
 error_code RdbSerializer::SendFullSyncCut() {
-  return WriteOpcode(RDB_OPCODE_FULLSYNC_END);
+  RETURN_ON_ERR(WriteOpcode(RDB_OPCODE_FULLSYNC_END));
+
+  // RDB_OPCODE_FULLSYNC_END followed by 8 bytes of 0.
+  // The reason for this is that some opcodes require to have at least 8 bytes of data
+  // in the read buffer (EnsureRead) and since RDB_OPCODE_FULLSYNC_END is one of
+  // the last opcodes, we respect this requirement by sending 8 bytes of 0.
```
Comment (adiholden) [inline src/server/rdb_save.cc:674]: Please write
one of the last opcodes sent to replica

File: src/server/rdb_save.cc [hunk 2]
```diff
@@ -666,7 +666,14 @@ error_code RdbSerializer::SaveStreamConsumers(streamCG* cg) {
 }
 
 error_code RdbSerializer::SendFullSyncCut() {
-  return WriteOpcode(RDB_OPCODE_FULLSYNC_END);
+  RETURN_ON_ERR(WriteOpcode(RDB_OPCODE_FULLSYNC_END));
+
+  // RDB_OPCODE_FULLSYNC_END followed by 8 bytes of 0.
+  // The reason for this is that some opcodes require to have at least 8 bytes of data
+  // in the read buffer (EnsureRead) and since RDB_OPCODE_FULLSYNC_END is one of
```
Comment (adiholden) [inline src/server/rdb_save.cc:673]: some opcodes are expected to have at least 8 bytes of data in the read buffer when consuming the rdb data

**Review by adiholden — APPROVED** [review]

File: src/server/rdb_save.cc (hunk 2 above)
Comment (adiholden) [reply src/server/rdb_save.cc:673]: ?

Comment (romange) [reply src/server/rdb_save.cc]: done

**Pull request merged**